### ISSUE TITLE
mullti_scan "Error: xcresult bundle not found" because merged xcresult file is now in subfolder

### ISSUE THREAD

**Issue by jcharr1** [lyndsey-ferguson/fastlane-plugin-test_center#230]
### New Issue Checklist

- [x] Updated `fastlane-plugin-test_center` to the latest version
- [x] I read the [README.md](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md)
- [x] I reviewed the [example(s)](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md) for the action(s) I am using
- [x] I have removed any sensitive data such as passwords, authentication tokens, or anything else I do not want to world to see

> If you love this fastlane plugin, consider sponsoring it or asking your company to sponsor it. I would really appreciate any
> gesture: https://github.com/sponsors/lyndsey-ferguson.

### Issue Description

I'm running tests in parallel batches and am suddenly getting this error after all the test runs finish but before multi_scan finishes. Looking at the console output, the merged results (report.test_result/junit/xcresult) from the parallel runs are being put in a subfolder named after the test suite instead of being put into my root `test_output` directory. You can see in the output that the merged files were placed in the `InnovativeUITests` subfolder but multi_scan seems to also be looking for the xcresults file one directory up where it used to be or something.

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>[17:37:19]: $ xcrun xcresulttool merge /Volumes/DATA/My\ Documents/ILL/iPhone/Projects/ILL\ for\ iOS/git/IL101\ for\ iOS/InnovativeLanguage.com/fastlane/test_output/InnovativeUITests-batch-3/report.xcresult /Volumes/DATA/My\ Documents/ILL/iPhone/Projects/ILL\ for\ iOS/git/IL101\ for\ iOS/InnovativeLanguage.com/fastlane/test_output/InnovativeUITests-batch-2/report.xcresult /Volumes/DATA/My\ Documents/ILL/iPhone/Projects/ILL\ for\ iOS/git/IL101\ for\ iOS/InnovativeLanguage.com/fastlane/test_output/InnovativeUITests-batch-1/report.xcresult --output-path /var/folders/dk/z57xpvj52nx4wdtb5zzw_q7c0000gr/T/collated_result_20200609-13257-bck45r.xcresult
[17:37:34]: ▸ [v3] Merged to: /var/folders/dk/z57xpvj52nx4wdtb5zzw_q7c0000gr/T/collated_result_20200609-13257-bck45r.xcresult
[17:37:35]: Finished collating xcresults to '/Volumes/DATA/My Documents/ILL/iPhone/Projects/ILL for iOS/git/IL101 for iOS/InnovativeLanguage.com/fastlane/test_output/InnovativeUITests/report.xcresult'
[17:37:35]: Error setting value '["/Volumes/DATA/My Documents/ILL/iPhone/Projects/ILL for iOS/git/IL101 for iOS/InnovativeLanguage.com/fastlane/test_output/InnovativeUITests/report.xcresult", "/Volumes/DATA/My Documents/ILL/iPhone/Projects/ILL for iOS/git/IL101 for iOS/InnovativeLanguage.com/fastlane/test_output/report.xcresult"]' for option 'xcresults'
</pre>
</details>

### Environment

<details>
  <pre><details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.15.5                                                        |
| Ruby                        | 2.6.3                                                          |
| Bundler?                    | true                                                           |
| Git                         | git version 2.24.3 (Apple Git-128)                             |
| Installation Source         | ~/.gem/bin/fastlane                                            |
| Host                        | Mac OS X 10.15.5 (19F101)                                      |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib |
| OpenSSL Version             | LibreSSL 2.8.3                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 11.5                                                           |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.149.1 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                       | Version | Update-Status |
| ---------------------------- | ------- | ------------- |
| fastlane-plugin-test_center  | 3.10.3  | ✅ Up-To-Date  |
| fastlane-plugin-xchtmlreport | 0.1.1   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                          | Version      |
| ---------------------------- | ------------ |
| bundler                      | 2.1.4        |
| CFPropertyList               | 3.0.2        |
| concurrent-ruby              | 1.1.6        |
| i18n                         | 0.9.5        |
| minitest                     | 5.14.1       |
| thread_safe                  | 0.3.6        |
| tzinfo                       | 1.2.7        |
| activesupport                | 4.2.11.3     |
| public_suffix                | 2.0.5        |
| addressable                  | 2.7.0        |
| httpclient                   | 2.8.3        |
| json                         | 2.3.0        |
| algoliasearch                | 1.27.3       |
| atomos                       | 0.1.3        |
| aws-eventstream              | 1.1.0        |
| aws-partitions               | 1.327.0      |
| aws-sigv4                    | 1.1.4        |
| jmespath                     | 1.4.0        |
| aws-sdk-core                 | 3.98.0       |
| aws-sdk-kms                  | 1.33.0       |
| aws-sdk-s3                   | 1.67.1       |
| babosa                       | 1.0.3        |
| claide                       | 1.0.3        |
| fuzzy_match                  | 2.0.4        |
| nap                          | 1.1.0        |
| netrc                        | 0.11.0       |
| ffi                          | 1.13.1       |
| ethon                        | 0.12.0       |
| typhoeus                     | 1.4.0        |
| cocoapods-core               | 1.9.3        |
| cocoapods-deintegrate        | 1.0.4        |
| cocoapods-downloader         | 1.3.0        |
| cocoapods-plugins            | 1.0.0        |
| cocoapods-search             | 1.0.0        |
| cocoapods-stats              | 1.1.0        |
| cocoapods-trunk              | 1.5.0        |
| cocoapods-try                | 1.2.0        |
| colored2                     | 3.1.2        |
| escape                       | 0.0.4        |
| fourflusher                  | 2.3.1        |
| gh_inspector                 | 1.1.3        |
| molinillo                    | 0.6.6        |
| ruby-macho                   | 1.4.0        |
| nanaimo                      | 0.2.6        |
| xcodeproj                    | 1.16.0       |
| cocoapods                    | 1.9.3        |
| colored                      | 1.2          |
| colorize                     | 0.8.1        |
| highline                     | 1.7.10       |
| commander-fastlane           | 4.4.6        |
| declarative                  | 0.0.10       |
| declarative-option           | 0.1.0        |
| digest-crc                   | 0.5.1        |
| unf_ext                      | 0.0.7.7      |
| unf                          | 0.1.4        |
| domain_name                  | 0.5.20190701 |
| dotenv                       | 2.7.5        |
| emoji_regex                  | 1.0.1        |
| excon                        | 0.73.0       |
| multipart-post               | 2.0.0        |
| faraday                      | 1.0.1        |
| http-cookie                  | 1.0.3        |
| faraday-cookie_jar           | 0.0.6        |
| faraday_middleware           | 1.0.0        |
| fastimage                    | 2.1.7        |
| jwt                          | 2.1.0        |
| memoist                      | 0.16.2       |
| multi_json                   | 1.14.1       |
| os                           | 1.1.0        |
| signet                       | 0.14.0       |
| googleauth                   | 0.12.0       |
| mini_mime                    | 1.0.2        |
| uber                         | 0.1.0        |
| representable                | 3.0.4        |
| retriable                    | 3.1.2        |
| google-api-client            | 0.38.0       |
| google-cloud-env             | 1.3.2        |
| google-cloud-errors          | 1.0.1        |
| google-cloud-core            | 1.5.0        |
| google-cloud-storage         | 1.26.2       |
| mini_magick                  | 4.10.1       |
| multi_xml                    | 0.6.0        |
| plist                        | 3.5.0        |
| rubyzip                      | 1.3.0        |
| security                     | 0.1.3        |
| naturally                    | 2.2.0        |
| simctl                       | 1.6.8        |
| slack-notifier               | 2.3.2        |
| terminal-notifier            | 2.0.0        |
| unicode-display_width        | 1.7.0        |
| terminal-table               | 1.8.0        |
| tty-screen                   | 0.8.0        |
| tty-cursor                   | 0.7.1        |
| tty-spinner                  | 0.9.3        |
| word_wrap                    | 1.0.0        |
| rouge                        | 2.0.7        |
| xcpretty                     | 0.3.0        |
| xcpretty-travis-formatter    | 1.0.0        |
| xctest_list                  | 1.1.8        |
| fastlane-plugin-test_center  | 3.10.3       |
| fastlane-plugin-xchtmlreport | 0.1.1        |
| xcresult                     | 0.2.1        |
| xcov                         | 1.7.3        |
</details>

*generated on:* **2020-06-09**
</details></pre>
</details>

**Comment by lyndsey-ferguson** [comment]
Thanks for the bug report @jcharr1, good to see you again even under these circumstances. 

Does `multi_scan` generate the same error if you use batches and no parallel runners? What about no batches and no parallel runners?

Could you attach a log generated by running your fastlane with the `--verbose` flag? My time is more limited these days so whatever you can do to help would be appreciated!

**Comment by jcharr1** [comment]
Hi @lyndsey-ferguson. I hope you and yours are doing ok. We're still kickin' over here so far.

I get the same error if I use batches and no parallel runner but not if I don't use either. Here are verbose logs for the 3 configs I tried.

[multi_scan-logs.zip](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/files/4755474/multi_scan-logs.zip)

**Comment by lyndsey-ferguson** [comment]
Okay. So it is not failing for me. Based on your logs, I guessed that maybe spaces in file paths contributed to the problem so I changed my `:output_directory` to have spaces. No change.

Can you post a sample `Fastfile`? Maybe you can modify the sample "AtomicBoy" project I have for this repo?

**Comment by lyndsey-ferguson** [comment]
@jcharr1 is there anyway that you can provide a sample?

**Comment by jcharr1** [comment]
@lyndsey-ferguson sorry, I must have missed the notification email for your reply the other day. This is a commercial app, so I have to be very careful how much of the project files I share. I'll do some more testing on my end and try out the sample project. I'll let you know how it goes and work on figuring how much I can share (the vast majority of the Fastfile should be ok).

**Comment by lyndsey-ferguson** [comment]
I totally understand. The main thing I need is something that I can use to reproduce the error. I suggested the sample hoping that you could tweak it to be similar enough to replicate the problem. 

Thanks!

**Comment by jcharr1** [comment]
I'm still poking around, but so far I've found that this starts happening for me in v3.10.3 of the plugin. If I downgrade to 3.10.2 this problem doesn't happen and all the test result files end up in my root test_output folder.

**Comment by lyndsey-ferguson** [comment]
I imagine that the problem is in [here](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/compare/v3.10.2...v3.10.3#diff-8e5624a2717b59358a99a62cd65433a3R215-R247)

**Comment by lyndsey-ferguson** [comment]
@jcharr1 I _may_ have fixed it while looking at another Issue. While I was making progress on their Issue, they faced the same crash. Do you want to try using the fix that I set up for them? 

If so, please modify your `Pluginfile` per my instructions below, run `bundle install`, and then run your fastlane again (with the `--verbose` flag)?

Pluginfile:
```
gem 'fastlane-plugin-test_center', :git => "https://github.com/lyndsey-ferguson/fastlane-plugin-test_center.git", :branch => "issue-233-prepopulated-devices-sent-to-scan"
```

If there are still problems, please let me know and attach the console output as a text file (removing any sensitive data) to this issue (makes it easier for me to review).  If it works, please let me know.

**Comment by jcharr1** [comment]
@lyndsey-ferguson That branch fixed it for me. Thanks. 😀

**Comment by gadireddi226** [comment]
@lyndsey-ferguson It fixed it for me too. Thanks for good work ! It will be probably included in v v3.11.3, right ?

**Comment by lyndsey-ferguson** [comment]
@jcharr1 @gadireddi226 I'm still waiting for the other bug reporter to test that issue's fix.  I've made a separate PR for that, can you test it? If it works, I'll merge and release it.

Can you modify your `Pluginfile` per my instructions below, run `bundle install`, and then run your fastlane again (with the `--verbose` flag)?

Pluginfile:
```
gem 'fastlane-plugin-test_center', :git => "https://github.com/lyndsey-ferguson/fastlane-plugin-test_center.git", :branch => "issue-230-xcresult-not-found-crash"
```

If there are still problems, please let me know and attach the console output as a text file to this issue (makes it easier for me to review).  If it works, please let me know.

**Comment by gadireddi226** [comment]
Tested the branch => "issue-230-xcresult-not-found-crash" . I had both problems(issue-230-xcresult-not-found-crash and issue-233-prepopulated-devices-sent-to-scan) and now everything works fine.

**Comment by lyndsey-ferguson** [comment]
Ok, great, I'll merge and release this so you no longer have the crash!

**Comment by lyndsey-ferguson** [comment]
Fixed in [v3.11.3](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/releases/tag/v3.11.3)
